pages/[detailPictureView].js

```diff
@@ -10,19 +10,43 @@ export default function DetailPictureViewPage({ countries, country }) {
   }
 
   const { publicIds, imagesUrls } = country;
-  const publicId = publicIds.findIndex((id) => id === currentPublicId);
+  const publicIdIndex = publicIds.findIndex((id) => id === currentPublicId);
 
-  const currentPicture = imagesUrls[publicId];
+  const currentPicture = imagesUrls[publicIdIndex];
 
   if (currentPicture === undefined) {
     return null;
   }
+
+  async function handleDeletePicture() {
+    const response = await fetch(`/api/${currentPublicId}`, {
+      method: "DELETE",
+    });
+
+    router.back();
+
+    const updatedPublicIds = country.publicIds.filter(
+      (id) => id !== currentPublicId
+    );
+    const updatedImageUrls = country.imagesUrls.filter(
+      (url) => url !== currentPicture
+    );
+    const updatedCountry = {
+      ...country,
+      publicIds: updatedPublicIds,
+      imagesUrls: updatedImageUrls,
+    };
+    deleteImage(updatedCountry);
```

Comment:
maybe you can move this code inside the "deleteImage" function and pass the currentPublicId and currenPicture as inputs to that function. That would be a better separation of concerns (the code for deleting the Image is entirely included in the deleteImage function)